src/controller/datacontroller.js

```diff
@@ -236,7 +237,19 @@ export default class DataController {
 	 * @returns {module:engine/model/documentfragment~DocumentFragment} Output document fragment.
 	 */
 	toModel( viewElementOrFragment, context = '$root' ) {
-		return this.viewToModel.convert( viewElementOrFragment, { context: [ context ] } );
+		const { conversionResult, markers } = this.viewToModel.convert( viewElementOrFragment, { context: [ context ] } );
```

Comment:
I think `convert` should simply return `documentFragmet` with markers which can be added instead of transforming it here. `DataController` should not contain any transformation logic, it should be as simple as possible.